app/views/installations/integration_providers/_integration.html.haml

```diff
@@ -0,0 +1,19 @@
+.col-lg-6
+  .panel.panel-default.integrations
+    .panel-heading
+      =provider.capitalize + " Integration"
+    .panel-body
+      - integration = @website.integrations.find_by_provider(provider)
+      - if integration.nil?
+        = button_to auths_path(integration: { name: provider.capitalize, provider: provider } ), class: "btn btn-default btn-block provider-button #{integration.try(:disabled) ? 'hidden' : ''}" do
+          %span{class: "#{provider}-button"}
+          = "Integrate " + provider.capitalize
+      - elsif integration.configuration["selected_application"].nil?
```

Comment:
You should make a method in base_driver for this. It's not safe to add this kind of checks here. We might change the data and forget to update this.